src/views/product-downloads-view/helpers.ts

```diff
@@ -261,6 +262,50 @@ export function prettyOs(os: string): string {
 	}
 }
 
+/**
+ * Given a version string,
+ * Return `true` if the string is a valid version with the build ID `"ent"`,
+ * or `false` if the string is a valid version not identified as enterprise,
+ * or `null` if the string could not be parsed with `semverParse`.
+ * or `false` otherwise.
+ *
+ * Note: returns `null` for invalid semver versions.
+ *
+ * Warning: some of our version numbers might not follow semver in the way
+ * we'd expect. As an example, our Consul version data seems to contain varying
+ * formats for "enterprise" version numbers, and certain pre-release formats
+ * will not be recognized as "ent" builds due to their formatting.
+ *
+ * Here are some specific examples from Consul releases:
+ * - `1.12.0-beta1+ent` will be parsed, as we might expect, as:
+ *   build: [ "ent" ]
+ *   prerelease: [ "beta1" ]
+ * - `1.10.0+ent-alpha` will be parsed, perhaps unexpectedly, as:
+ *   build: [ "ent-alpha" ]
+ *   prerelease: []
+ *
+ * The latter example is a valid version according to semver, but might not match our
+ * expectations. The build ID is being included before the pre-release ID,
+ * which Semver sees as a build ID with a dash in it. While we could in theory
+ * try to parse out [ "alpha" ] pre-release and [ "ent" ] build IDs, this
+ * could be brittle if we ever want a build ID that does contain dashes.
+ *
+ * A more robust path forward would be to fix the versioning format at the
+ * source to match the semver spec, which specifies that the build ID should
+ * always be after the pre-release ID.
+ *
+ * Spec: https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions
+ * List of Consul releases: https://releases.hashicorp.com/consul/index.json
+ */
+export function getIsEnterpriseVersion(version: string): boolean | null {
```

Comment:
I've replaced the previous regex-based approach with `semverParse`. There are some quirks here I thought might be worth noting, that are likely related to the fix in this PR for ensuring pre-releases don't show up in the enterprise downloads page dropdowns.

| Before | After |
| - | - |
| <img width="1489" alt="before-ent-pre-releases-shown" src="https://user-images.githubusercontent.com/4624598/223864589-8d6c59cb-411f-4f7b-8b7e-78256c3a7385.png"> | <img width="1491" alt="after-ent-pre-releases-not-shown" src="https://user-images.githubusercontent.com/4624598/223864594-78702cb7-7981-49e2-a3d2-47ebc710c5f0.png"> |

Reply:
wow, nice fix!